[CI] clang-tidy pipeline failed

### Description

https://github.com/apache/gluten/actions/runs/25201801600/job/73894188462

```
fatal: detected dubious ownership in repository at '/__w/gluten/gluten'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/gluten/gluten
fatal: detected dubious ownership in repository at '/__w/gluten/gluten'
To add an exception for this directory, call:

	git config --global --add safe.directory /__w/gluten/gluten
Error: Process completed with exit code 128.
```

### Gluten version

None

--- a/.github/workflows/cpp_clang_tidy.yml
+++ b/.github/workflows/cpp_clang_tidy.yml
@@ -38,6 +38,10 @@ jobs:
         with:
           ref: ${{ github.event.workflow_run.head_sha }}
           fetch-depth: 0
+      # Workspaces bind-mounted into containers are often owned by a different UID than the
+      # container user; Git 2.35+ refuses operations unless the directory is marked safe.
+      - name: Configure Git safe directory (container)
+        run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
       - name: Detect C++ file changes
         id: filter
         run: |

